packages/teams-js/prepNextDevRelease.js

```diff
@@ -86,84 +86,75 @@ function getPrefix(version) {
   }
 }
 
+/**
+ * Takes the given whole version number and gets the suffix number of the version type (e.g. 'beta').
+ * @param {string} versionType The type of the version number suffix to get. e.g. beta, dev
+ * @param {string} wholeVerNum The entire version number. e.g. 2.0.0-beta.0, 2.0.0-beta.1-dev.0
+ * @returns The suffix number of the version type in the whole version number.
+ */
+function getSpecificVerSuffixNum(versionType, wholeVerNum) {
+  const indexOfVerType = wholeVerNum.indexOf(versionType);
+  if (indexOfVerType === -1) {
+    return -1;
+  } else {
+    return parseInt(wholeVerNum.slice(indexOfVerType + versionType.length + '.'.length));
+  }
+}
+
 /**
  * Uses the given current next-dev version and latest production version to generate and return
  * the number of the new next-dev version. The new next-dev version numbers are 0-index based.
- * @param {string} devVer The version currently tagged next-dev.
- * @param {string} currVer The version taken from the package.json.
- * @returns Just the number of the suffix of the next next-dev version. i.e. 1.10.0-dev.<next-dev suffix number to be returned>
+ * @param {string} currNextDevVer The version currently tagged next-dev. (e.g. 2.0.0-beta.1-dev.1)
+ * @param {string} currPkgJsonVer The version taken from the package.json. (e.g. 2.0.0-beta.0)
+ * @returns Just the number of the suffix of the new next-dev version number. (e.g. return 2 if next next-dev version is 2.0.0-beta.1-dev.2)
  */
-function getDevSuffixNum(devVer, currVer) {
-  if (devVer === undefined) {
+function getNewNextDevSuffixNum(currNextDevVer, currPkgJsonVer) {
+  if (currNextDevVer === undefined || currNextDevVer === '') {
     return 0;
   }
 
-  const [major, minor, patch] = currVer.split('.');
-  const [devMajor, devMinor, devPatch] = getPrefix(devVer).split('.');
-
-  if (parseInt(devMajor) < parseInt(major)) {
-    return 0;
-  } else if (devMajor == major) {
-    if (parseInt(devMinor) <= parseInt(minor)) {
-      return 0;
-    }
+  const betaSuffixNumInCurrNextDev = getSpecificVerSuffixNum('beta', currNextDevVer);
+  const betaSuffixNumInPkgJsonVer = getSpecificVerSuffixNum('beta', currPkgJsonVer);
+  // The new next-dev suffix number will be 0 if the beta suffixes are equal.
+  let newNextDevVerSuffixNum = 0;
+
+  // If the beta suffix number in the current next-dev version is already bumped from the beta suffix number of the package.json version,
+  // increment the existing dev suffix number from the next-dev version.
+  if (betaSuffixNumInCurrNextDev === betaSuffixNumInPkgJsonVer + 1) {
+    newNextDevVerSuffixNum = getSpecificVerSuffixNum('dev', currNextDevVer) + 1;
+    // If the next-dev beta suffix is not equal to or exactly 1 higher than the package.json beta suffix, throw an error since this should not happen.
```

Comment:
>// If the next-dev beta suffix is not equal to or exactly 1 higher than the package.json beta suffix, throw an error since this should not happen. [](http://example.com/codeflow?start=4&length=146)

An example here would also be great :)

Reply:
Done!